core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -135,7 +134,9 @@ public FSDataOutputStream append(Path path, int bufferSize, Progressable progres
       if (mFileSystem.exists(uri)) {
         throw new IOException(ExceptionMessage.FILE_ALREADY_EXISTS.getMessage(uri));
       }
-      return new FSDataOutputStream(mFileSystem.createFile(uri), mStatistics);
+      return new FSDataOutputStream(
+          mFileSystem.createFile(uri, CreateFilePOptions.newBuilder().setRecursive(true).build()),
```

Comment:
why is recursive set now?

Reply:
Alluxio Java API change. Explained in other comment.